kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/impl/JavaBackedType.java

```diff
@@ -0,0 +1,49 @@
+package org.kie.dmn.feel.lang.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.kie.dmn.feel.lang.CustomType;
+import org.kie.dmn.feel.lang.FEELAccessor;
+import org.kie.dmn.feel.lang.Type;
+
+public class JavaBackedType implements CustomType {
+    
+    private List<Field> fields = new ArrayList<>();
+    private Class<?> wrapped;
+
+    public JavaBackedType(Class<?> class1) {
+        this.wrapped = class1;
+        Stream.of( class1.getMethods() )
+            .filter( m -> m.getAnnotation(FEELAccessor.class) != null )
```

Comment:
Not sure if I was clear when we discussed this, but the annotation should be optional. If there is no annotation, we should still support the java class as a type, just that the name of the property will be based on the name of the javabean method, not on a possible defined name in the annotation itself.